pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -125,7 +140,7 @@ func (pl *DefaultPreemption) preempt(ctx context.Context, state *framework.Cycle
 	}
 
 	// 2) Find all preemption candidates.
-	candidates, err := FindCandidates(ctx, cs, state, pod, m, ph, nodeLister, pl.pdbLister)
+	candidates, err := FindCandidates(ctx, cs, state, pod, m, ph, pl, nodeLister, pl.pdbLister)
```

Comment:
Can we hold passing the whole `pl` to `FindCandidates`? `pl` here represents the whole DefaultPreemption struct.

An alternative is to make `candidateOffset` a pointer, and `args` needs to be passed as well.

Reply:
+1. Passing the whole Plugin would make the function unusable for custom plugins.

-1 to Passing args. This is the configuration struct for this particular plugin. Custom plugins shouldn't have to import the internal config package. It think we should use the options pattern. `WithCandidateOffset`, `WithMaxPercentage`, etc.

Reply:
> -1 to Passing args. This is the configuration struct for this particular plugin. Custom plugins shouldn't have to import the internal config package.

Good point.